packages/contracts-bedrock/contracts/EAS/EAS.sol

```diff
@@ -0,0 +1,853 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.19;
+
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import { Semver } from "../universal/Semver.sol";
+
+import { EIP712Verifier } from "./eip712/EIP712Verifier.sol";
+
+import { ISchemaResolver } from "./resolver/ISchemaResolver.sol";
+
+// prettier-ignore
+import {
+    AccessDenied,
+    EMPTY_UID,
+    EIP712Signature,
+    InvalidLength,
+    MAX_GAP,
+    NotFound,
+    NO_EXPIRATION_TIME,
+    uncheckedInc
+} from "./Common.sol";
+
+// prettier-ignore
+import {
+    Attestation,
+    AttestationRequest,
+    AttestationRequestData,
+    DelegatedAttestationRequest,
+    DelegatedRevocationRequest,
+    IEAS,
+    MultiAttestationRequest,
+    MultiDelegatedAttestationRequest,
+    MultiDelegatedRevocationRequest,
+    MultiRevocationRequest,
+    RevocationRequest,
+    RevocationRequestData
+} from "./IEAS.sol";
+
+import { ISchemaRegistry, SchemaRecord } from "./ISchemaRegistry.sol";
+
+struct AttestationsResult {
+    uint256 usedValue; // Total ETH amount that was sent to resolvers.
+    bytes32[] uids; // UIDs of the new attestations.
+}
+
+/**
+ * @title EAS - Ethereum Attestation Service
+ */
+contract EAS is IEAS, Semver, Initializable, EIP712Verifier {
+    using Address for address payable;
+
+    error AlreadyRevoked();
+    error AlreadyRevokedOffchain();
+    error AlreadyTimestamped();
+    error InsufficientValue();
+    error InvalidAttestation();
+    error InvalidAttestations();
+    error InvalidExpirationTime();
+    error InvalidOffset();
+    error InvalidRegistry();
+    error InvalidRevocation();
+    error InvalidRevocations();
+    error InvalidSchema();
+    error InvalidVerifier();
+    error Irrevocable();
+    error NotPayable();
+    error WrongSchema();
+
+    // The global schema registry.
+    ISchemaRegistry private immutable _schemaRegistry;
+
+    // The global mapping between attestations and their UIDs.
+    mapping(bytes32 uid => Attestation attestation) private _db;
+
+    // The global mapping between data and their timestamps.
+    mapping(bytes32 data => uint64 timestamp) private _timestamps;
+
+    // The global mapping between data and their revocation timestamps.
+    mapping(address revoker => mapping(bytes32 data => uint64 timestamp)) private _revocationsOffchain;
+
+    // Upgrade forward-compatibility storage gap
+    uint256[MAX_GAP - 3] private __gap;
+
+    /**
+     * @dev Creates a new EAS instance.
+     *
+     * @param registry The address of the global schema registry.
+     */
+    constructor(ISchemaRegistry registry) Semver(1, 0, 0) EIP712Verifier("EAS", "1.0.0") {
```

Comment:
The registry can be removed as an argument and the hardcoded predeploy constant can be used, see my comment in `SchemaRegistry`